knock

```diff
@@ -30,22 +30,33 @@ class Knocker(object):
         self.timeout = args.timeout / 1000
         self.delay = args.delay / 1000
         self.use_udp = args.udp
-        self.host = args.host
-        self.is_ipv6 = args.host.find(':') != -1
         self.ports = args.ports
 
+        self.address_family, self.socket_type, _, _, socket_address = socket.getaddrinfo(
+                host=args.host,
+                port=None,
+                type=0,
+                proto=socket.IPPROTO_UDP if self.use_udp else socket.IPPROTO_TCP,
+                flags=socket.AI_ADDRCONFIG
+            )[0]
+
+        self.socket_address = list(socket_address)
+
     def knock(self):
         last_index = len(self.ports) - 1
         for i, port in enumerate(self.ports):
-            s = socket.socket(
-                socket.AF_INET6 if self.is_ipv6 else socket.AF_INET,
-                socket.SOCK_DGRAM if self.use_udp else socket.SOCK_STREAM
-            )
+            socket_address = self.socket_address
+            socket_address[1] = port
+            socket_address = tuple(socket_address)
+
+            s = socket.socket(self.address_family, self.socket_type)
             s.setblocking(False)
-            s.connect_ex((self.host, port))
-            select.select([s], [s], [s], self.timeout)
+
             if self.use_udp:
-                s.send(b'')
+                s.sendto(b'', socket_address)
```

Comment:
Just tested and received the following error. I'm investigating, but sharing in case you have any immediate insight:
```
C:\Users\martin\Desktop>knock.py -u ipv6.netconfigs.com 12345
Traceback (most recent call last):
  File "C:\Users\martin\Desktop\knock.py", line 68, in <module>
    Knocker(sys.argv[1:]).knock()
  File "C:\Users\martin\Desktop\knock.py", line 56, in knock
    s.sendto(b'', socket_address)
OSError: [WinError 10057] A request to send or receive data was disallowed because the socket is not connected and (when sending on a datagram socket using a sendto call) no address was supplied

C:\Users\martin\Desktop>
```